psycopg_pool/psycopg_pool/pool_async.py

```diff
@@ -40,10 +40,10 @@ class AsyncConnectionPool(Generic[ACT], BasePool):
 
     def __init__(
         self,
-        conninfo: str = "",
+        conninfo: str | Callable[[], Awaitable[str]] | None = None,
```

Comment:
In these signatures we are not considering the case of an async connection using a sync conninfo/kwargs function. Using the proposed typedefs this would be supported, and `ensure_async()` will do the right thing.